console/EAP7-1156_Display_open_network_ports_for_a_managed_server.adoc

```diff
@@ -0,0 +1,49 @@
+= EAP7-1156 Display open network ports for a managed server
+:author:            Claudio Miranda
+:email:             claudio@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+When users wants to list the network open ports of a managed server, they can open HAL model browser and navigate to the socket-bindings and open each socket-binding to show the bound-port, however this is not a productive method.
+This proposal is to list all bound ports and the socket-binding name on a preview pane per server.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/HAL-532
+* https://issues.jboss.org/browse/EAP7-1156
+
+=== Dev Contacts
+
+* mailto:claudio@redhat.com[Claudio Miranda]
+
+=== QE Contacts
+
+* mailto:spriadka@redhat.com[Šimon Priadka]
+
+== Requirements
+
+To show the bound ports, the following operation is to be used.
+
+`/socket-binding-group=standard-sockets/socket-binding=http:read-resource(include-runtime)`
+
+* Show all bound ports of each socket-binding and its name.
+* Show them in the server preview pane.
+* Show the open ports when the server is in the following status: running, reload-required, restart-required
+* Do not show while the server is starting, stopping or in suspended mode.
+* Should work in domain and standalone mode.
+
+=== Non-Requirements
+
+There are services that doesn't register themselves with the socket-binding, as a consequence the socket is open but it is not displayed as bound, they will not be displayed in the "Opent Ports" preview. Currently `modcluster` is one service that doesn't register the socket-binding.
```

Comment:
If a consumer of a socket-binding doesn't record that it has bound the socket, that's a bug. I agree though that it should not be a requirement of the implementors of this RFE to correct such bugs.